recipes/hwloc/all/conanfile.py

```diff
@@ -0,0 +1,118 @@
+from conan import ConanFile
+from conan.tools.apple import is_apple_os, fix_apple_shared_install_name
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, rm, rmdir
+from conan.tools.gnu import Autotools, AutotoolsDeps, AutotoolsToolchain
+from conan.tools.layout import basic_layout
+import os
+
+required_conan_version = ">=1.53.0"
+
+
+class HwlocConan(ConanFile):
+    name = "hwloc"
+    description = "Portable Hardware Locality (hwloc)"
+    topics = ("hwloc")
+    license = "BSD-3-Clause"
+    homepage = "https://www.open-mpi.org/projects/hwloc/"
+    url = "https://github.com/conan-io/conan-center-index"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "disable_libxml2": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "disable_libxml2": True
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        if not self.options.disable_libxml2:
+            self.requires("libxml2/2.9.12")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def layout(self):
+        if self.settings.os == "Windows":
+            cmake_layout(self, src_folder="src")
+        else:
+            basic_layout(self, src_folder="src")
+
+    def generate(self):
+        if self.settings.os == "Windows":
+            deps = CMakeDeps(self)
+            deps.generate()
+            tc = CMakeToolchain(self)
+            tc.cache_variables["HWLOC_ENABLE_TESTING"] = 'OFF'
+            tc.cache_variables["HWLOC_SKIP_LSTOPO"] = 'ON'
+            tc.cache_variables["HWLOC_SKIP_TOOLS"] = 'ON'
+            tc.cache_variables["HWLOC_SKIP_INCLUDES"] = 'OFF'
+            tc.cache_variables["HWLOC_WITH_OPENCL"] = 'OFF'
+            tc.cache_variables["HWLOC_WITH_CUDA"] = 'OFF'
+            tc.cache_variables["HWLOC_BUILD_SHARED_LIBS"] = self.options.shared
+            tc.cache_variables["HWLOC_WITH_LIBXML2"] = not self.options.disable_libxml2
+            tc.generate()
+        else:
+            deps = AutotoolsDeps(self)
```

Comment:
AutotoolsDeps is not the best option for autotools dependencies, but ``PkgConfigDeps`` should be preferred in the general case. As the default doesn't include dependencies, it is difficult to know if this is working or not, did you test it locally? Please try to change it if possible.

Reply:
Applied comments.
Yes, I have tested with `with_libxml2=True/False` on macOS, Linux.